### ISSUE TITLE
XSqueeze on AKASO HM8 (ARM)

### ISSUE THREAD

**Issue by Tschipel** [bossanova808/repository.bossanova808#7]
Hello,
i tried to install the XSqueeze AddOn for LibreElec on an AKASO HM8 (ARM) device. Is there an issue in the plugin or is this LibreElec related (variables not set correctly?).

12:32:52.700 T:3745518496  NOTICE: ### XSqueeze - 1.2.6 ### XSqueeze (Author: bossanova808) Starting ...
12:32:52.701 T:3745518496  NOTICE: ### XSqueeze - 1.2.6 ### Called as: ['']
12:32:52.702 T:3745518496  NOTICE: ### XSqueeze - 1.2.6 ### Installed custom keymap
12:32:52.702 T:3745518496  NOTICE: ### XSqueeze - 1.2.6 ### Suspended XBMC AE
12:32:52.703 T:3745518496  NOTICE: ### XSqueeze - 1.2.6 ### Starting local player [squeezelite], system is [linux]
12:32:52.703 T:3745518496  NOTICE: ### XSqueeze - 1.2.6 ### Attempting to start player: ['/storage/.kodi/addons/script.xsqueeze/resources/bin/linux/squeezelite-i386', '-m', '00:00:00:00:00:01', '-s', '192.168.6.159']
12:32:52.703 T:3745518496  NOTICE: ### XSqueeze - 1.2.6 ### Path is: ['/storage/.kodi/addons/script.xsqueeze', '/storage/.kodi/addons/script.module.b808common/lib', '/storage/.kodi/addons/script.module.xsqueezecommon/lib', '/usr/lib/python27.zip', '/usr/lib/python2.7', '/usr/lib/python2.7/plat-linux2', '/usr/lib/python2.7/lib-tk', '/usr/lib/python2.7/lib-old', '/usr/lib/python2.7/lib-dynload', '/usr/lib/python2.7/site-packages', '/usr/lib/python2.7/site-packages/gtk-2.0', '/storage/.kodi/addons/script.xsqueeze/resources/lib/classes', '/storage/.kodi/addons/script.xsqueeze/resources/lib']
12:32:52.726 T:3745518496  NOTICE: ### XSqueeze - 1.2.6 ### Failed creating player process!
12:32:52.727 T:3745518496  NOTICE: ### XSqueeze-1.2.6 ### Exception:Traceback (most recent call last):
                                              File "/storage/.kodi/addons/script.xsqueeze/default.py", line 333, in <module>
                                                slaveProcess = subprocess.Popen(exe, shell=False)
                                              File "/usr/lib/python2.7/subprocess.py", line 390, in __init__
                                              File "/usr/lib/python2.7/subprocess.py", line 1024, in _execute_child
                                            OSError: [Errno 8] Exec format error

**Comment by bossanova808** [comment]
It's trying to use the normal linux binary and not the arm one....

I need a pastebin of a FULL debug log, and also ideally the output of `uname -a` on your system there

**Comment by Tschipel** [comment]
LibreELEC:~ # uname -a
Linux LibreELEC 3.14.29 #1 SMP Thu Feb 23 10:07:41 CET 2017 aarch64 GNU/Linux

Will create a full debug - please remain patient.

**Comment by Tschipel** [comment]
Hi bossanova808,
is the attached link with the full debug log enough?

**Comment by bossanova808** [comment]
ah it should be - sorry, it didn't alert me when you added the log so I hadn't seen this.

Bit busy at work - give me a few days.  In the meantime if you ssh in and copy/rename the arm binaries to the linux folder that would get you going at least....

**Comment by Tschipel** [comment]
No problem, just wanted to know if you've seen it.

Already copied and renamed the binaries, but this didn't work. Will do again later and provide the result(s)

**Comment by bossanova808** [comment]
hmm, ok, you might need to seek out a binary that works - been a long time since I looked at that/what's out there.  The arm binaries are Pi hard float ones, might be it needs a different arm one altogether...

**Comment by Tschipel** [comment]
Here the debug log with exchanged binary (arm moved and renamed to linux)
http://sprunge.us/ROaL

**Comment by bossanova808** [comment]
Yeah I really need what happens when you ssh in and try and run the binary directly...

**Comment by Tschipel** [comment]
LibreELEC:~ # /storage/.kodi/addons/script.xsqueeze/resources/bin/linux/squeezel
ite-i386', '-m', '00:00:00:00:00:01', '-s', '192.168.6.159'
>

**Comment by bossanova808** [comment]
err, no, that's not right.  You need to manually run the command line version of that e.g.:

`cd /storage/.kodi/addons/script.xsqueeze/resources/bin/linux/`
`./squeezelite -m 00:00:00:00:00:01 -s 192.168.6.159`

**Comment by Tschipel** [comment]
login as: root
root@192.168.6.83's password:
##############################################
#                 LibreELEC                  #
#            http://libreelec.tv             #
##############################################

LibreELEC (community) Version: 8.0.0d
LibreELEC git: 35760edfb62ee9c2f4134fd0f53f1612d3514c22
LibreELEC:~ # cd /storage/.kodi/addons/script.xsqueeze/resources/bin/linux/
LibreELEC:~/.kodi/addons/script.xsqueeze/resources/bin/linux # ./squeezelite -m
00:00:00:00:00:01 -s 192.168.6.159
[11:09:15.512364] test_open:124 playback open error: Device or resource busy
[11:09:15.513424] output_init_common:373 unable to open output device
LibreELEC:~/.kodi/addons/script.xsqueeze/resources/bin/linux #

**Comment by bossanova808** [comment]
Yep so the binary is OK. Your audio device is being hogged by Kodi. You need to disable gui sounds, or keep Aaudio device alive etc. As per the wiki. The player is crashing because it can't open the audio device.

**Comment by Tschipel** [comment]
Strange - no GUI sounds active, but will check this evening again - no more time at the moment. Thanks! :)

**Comment by Tschipel** [comment]
OK, the binary works in linux directory. 

After several tests i finally figured out what the problem is/was here.

I had to change the XSqueeze -> Player Arguments -> Automatic SqueezeLite Output Discovery from default to the setting '-o hdmi:CARD=AMLM8AUDIO,DEV=0'

**Comment by bossanova808** [comment]
So to confirm - all is working (now that youre choosing an output) - without you needing to change the binaries?  It uses the linux ones by default..

I.e. I don't need to change anything?

**Comment by Tschipel** [comment]
Err no. I copied the arm binary to the linux folder and renamed it. To make it finally working i had to change the output from default to hdmi. So it only works using the 'backdoor' ;)

But many thanks for your support!

**Comment by bossanova808** [comment]
Ok, right, I'll have to look into how Kodi can more correctly tell me about your platform.

Glad you're going for now, will look into this when I have a bit more time - might need some more info from you but will checkout out recent Kodi changes in this area first....

**Comment by bossanova808** [comment]
Ok can you try something for me, this should solve it I think

Start with a clean install of XSqueeze (i.e. restore those binaries - I'd just uninstall and re-install it).

Then, you need to change some code in a module XSqueeze uses

this will be in addons/script.module.b808common/lib/b808common.py

around lines 280, insert a couple of lines after the `log("uname is: " + str(uname))` (you can ssh in and then use e.g. `nano` to do this I would think...or whatever editor LibreElec offers...or do it on another CP and SFTP it across)

Note that python is indentation sensitive so it must be _exactly_ as it appears here:

```
#log the detemined system type
log("uname is: " + str(uname))

if(uname[4].startswith("arm")):
    SYSTEM = "arm"

log("System is " + SYSTEM)
```

....and then run it again.  That should pick up you're using linux on arm and thus use the arm binaries in their normal spot.

Let me know, if all good I'll make the change and push a fix.

Thanks!

**Comment by Tschipel** [comment]
Ok, did as you requested. Now it works like a charm! :+1:

**Comment by bossanova808** [comment]
Fixed in 1.2.7
